components/shooter.py

```diff
@@ -0,0 +1,59 @@
+class Shooter:
+    GEAR_RATIO: float  # whatever this is
+
+    # top_flywheel: whatever motor type this is
+    # bottom flywheel: whatever motor type this is
+    # neck_motor: whatever motor this is <- this refers to the back motor that pulls the cube away from the motors
+
+    # loaded_switch Digital I/O <- in my head this has something to confirm when the shooter is correctly loaded
+
+    def __init__(self) -> None:
+        # create instances of hardware handles
+
+        # initialise motor speed
+        # initialise shooting
+        # initialise last shooting
+        # initialise loading
+        pass
+
+    def set_flywheel_speed(self) -> None:
+        # update internal motor speed setpoint
+        pass
+
+    def shoot(self) -> None:
+        # rotate back motors so the cube is picked up by the flywheels
+        # set loading variable
+        pass
+
+    def flywheel_error(self) -> float:
+        # return actual velocity - reference velocity
+        return 0.0
+
+    def load(self) -> None:
+        # set load variable
+        # set shooting variable
+        pass
+
+    def is_loaded(self) -> bool:
+        # return state of loaded switch
+        return False
+
+    def has_shot(self) -> bool:
+        # true if shooting and change time greater than threshold
```

Comment:
What edge are you envisioning being checked by this?

From a quick glance this sounds like it'll be keeping time as state. This seems a bit odd since surely that'd be better tracked by the state machine with a timed state.